src/Terminal.js

```diff
@@ -954,8 +954,38 @@ var Terminal = {
                 }
 				break;
 			case "top":
-				//TODO List each's script RAM usage
-                post("Not yet implemented");
+                if(commandArray.length != 1) {
+                  post("Incorrect usage of top command. Usage: top"); return;
+                }
+                
+                post("Script                    RAM Usage                 Threads");
+                var currRunningScripts = Player.getCurrentServer().runningScripts;
+                for(i = 0; i < currRunningScripts.length; i++) {
+                    //Check if script is on current server
+                    for(j = 0; j < Player.getCurrentServer().scripts.length; j++) {
+                        if(currRunningScripts[i] === Player.getCurrentServer().scripts[j].filename) {
+                            var script = Player.getCurrentServer().scripts[j];
+
+                            //Add script name
+                            var entry = script.filename;
```

Comment:
Instead of iteratively concatenating substrings, it is far more performant in JS to build up an array of substrings, and then to `.join()` them.